### ISSUE TITLE
Jenkins is not working well, login issues and cannot create pipeline via ks

### ISSUE THREAD

**Issue by LinuxSuRen** [kubesphere/kubesphere#3105]
**Describe the Bug**
Cannot log into Jenkins by default user and password when I just installed ks by [kk](https://github.com/kubesphere/kubekey). I cannot create the pipeline job on the DevOps project page as well.

**Versions Used**
KubeSphere: 3.0

**Environment**
None.

**How To Reproduce**
1. Go to the DevOps project page
2. You can see the exception stack, I write down the part of them below:

```
java.lang.NullPointerException at io.kubesphere.jenkins.devops.auth.KubesphereApiTokenAuthenticator.getReviewResponse(KubesphereApiTokenAuthenticator.java:94)
```

**Expected behavior**
Everything is fine no matter I log into Jenkins or create a pipeline job.

**Comment by LinuxSuRen** [comment]
Than @wansir for giving me a temporary solution that can resolve this issue. If anyone met the same problem above, you can try the following command:

`kubectl patch users admin -p '{"spec":{"password":"yourPassword"}}' --type='merge' && kubectl annotate users admin iam.kubesphere.io/password-encrypted-`

> Please don't forget to replace the `yourPassword` with your real one

**Comment by yasir2000** [comment]
> 
> 
> Than @wansir for giving me a temporary solution that can resolve this issue. If anyone met the same problem above, you can try the following command:
> 
> `kubectl patch users admin -p '{"spec":{"password":"admin"}}' --type='merge' && kubectl annotate users admin iam.kubesphere.io/password-encrypted-`
> 
> > Please don't forget to replace the `admin` with your real one

This command statement had encrypted my admin kubesphere password so now I cant access it

**Comment by LinuxSuRen** [comment]
@yasir2000 Happy to help you out of it.

**Comment by webup** [comment]
Will this bug fixed in upcoming 3.1 edition?

**Comment by LinuxSuRen** [comment]
hi @wansir , do you know about the fixing plan? I guess many users met this issue like @webup .

**Comment by wansir** [comment]
OpenLDAP is a stateful service that may be unavailable and cause this problem.  Plan to use [OIDC](https://plugins.jenkins.io/oic-auth/) instead of OpenLDAP in 3.1.

/milestone 3.1.0

**Comment by JohnNiang** [comment]
Same issue appeared just now installed by ks-installer: <https://hub.docker.com/layers/kubespheredev/ks-installer/latest/images/sha256-fc342b14415b0c326b92bd5dfeaed46c4fd23692ced979759218ff9575a7fd25?context=explore>

**Comment by JohnNiang** [comment]
Related Jenkins logs:

```java
 2021-04-25 02:19:55.909+0000 [id=26] INFO j.util.groovy.GroovyHookScript#execute: Executing /var/jenkins_home/init.groovy.d/initRBAC.groovy

Enabling role based authorisation strategy...

2021-04-25 02:20:01.484+0000 [id=26] INFO o.c.j.p.k.KubernetesClientProvider$SaveableListenerImpl#onChange: Invalidating Kubernetes client: kubernetes null

Admin role created...OK

Read role created...OK

Saving changes.

2021-04-25 02:20:01.705+0000 [id=26] INFO o.c.j.p.k.KubernetesClientProvider$SaveableListenerImpl#onChange: Invalidating Kubernetes client: kubernetes null

2021-04-25 02:20:01.705+0000 [id=26] INFO j.util.groovy.GroovyHookScript#execute: Executing /var/jenkins_home/init.groovy.d/initSonarqube.groovy

--> SonarQube Disabled

2021-04-25 02:20:04.091+0000 [id=26] INFO jenkins.InitReactorRunner$1#onAttained: Completed initialization

2021-04-25 02:20:04.105+0000 [id=20] INFO o.c.j.p.k.KubernetesClientProvider$SaveableListenerImpl#onChange: Invalidating Kubernetes client: kubernetes null

2021-04-25 02:20:04.216+0000 [id=20] INFO hudson.WebAppMain$3#run: Jenkins is fully up and running

2021-04-25 02:20:31.931+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.tasks.Maven.MavenInstaller

2021-04-25 02:20:32.405+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.plugins.sonar.MsBuildSonarQubeRunnerInstaller

2021-04-25 02:20:32.680+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.plugins.sonar.SonarRunnerInstaller

2021-04-25 02:20:32.919+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.tasks.Ant.AntInstaller

2021-04-25 02:20:35.587+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.tools.JDKInstaller

2021-04-25 02:20:35.588+0000 [id=57] INFO hudson.util.Retrier#start: Performed the action check updates server successfully at the attempt #1

2021-04-25 02:20:35.591+0000 [id=57] INFO hudson.model.AsyncPeriodicWork#lambda$doRun$0: Finished Download metadata. 45,499 ms

2021-04-25 02:20:46.220+0000 [id=16] SEVERE i.k.j.d.a.KubesphereApiTokenAuthenticator#authenticate: cannot get the review response or status is null by admin

2021-04-25 02:20:46.305+0000 [id=16] WARNING o.a.p.l.a.BindAuthenticator2#handleBindException: Failed to bind to LDAP: userDnuid=admin,ou=Users,dc=kubesphere,dc=io username=admin

javax.naming.AuthenticationException: [LDAP: error code 49 - Invalid Credentials]

at com.sun.jndi.ldap.LdapCtx.mapErrorCode(LdapCtx.java:3154)

at com.sun.jndi.ldap.LdapCtx.processReturnCode(LdapCtx.java:3100)

at com.sun.jndi.ldap.LdapCtx.processReturnCode(LdapCtx.java:2886)

at com.sun.jndi.ldap.LdapCtx.connect(LdapCtx.java:2800)

at com.sun.jndi.ldap.LdapCtx.<init>(LdapCtx.java:319)

at com.sun.jndi.ldap.LdapCtxFactory.getUsingURL(LdapCtxFactory.java:192)

at com.sun.jndi.ldap.LdapCtxFactory.getUsingURLs(LdapCtxFactory.java:210)

at com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxInstance(LdapCtxFactory.java:153)

at com.sun.jndi.ldap.LdapCtxFactory.getInitialContext(LdapCtxFactory.java:83)

at javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:684)

at javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:313)

at javax.naming.InitialContext.init(InitialContext.java:244)

at javax.naming.InitialContext.<init>(InitialContext.java:216)

at javax.naming.directory.InitialDirContext.<init>(InitialDirContext.java:101)

at org.acegisecurity.ldap.DefaultInitialDirContextFactory.connect(DefaultInitialDirContextFactory.java:180)

at org.acegisecurity.ldap.DefaultInitialDirContextFactory.newInitialDirContext(DefaultInitialDirContextFactory.java:261)

at org.acegisecurity.ldap.LdapTemplate.execute(LdapTemplate.java:123)

at org.acegisecurity.ldap.LdapTemplate.retrieveEntry(LdapTemplate.java:165)

at org.acegisecurity.providers.ldap.authenticator.BindAuthenticator.bindWithDn(BindAuthenticator.java:87)

at org.acegisecurity.providers.ldap.authenticator.BindAuthenticator.authenticate(BindAuthenticator.java:72)

at org.acegisecurity.providers.ldap.authenticator.BindAuthenticator2.authenticate(BindAuthenticator2.java:49)

at org.acegisecurity.providers.ldap.LdapAuthenticationProvider.retrieveUser(LdapAuthenticationProvider.java:233)

at org.acegisecurity.providers.dao.AbstractUserDetailsAuthenticationProvider.authenticate(AbstractUserDetailsAuthenticationProvider.java:122)

at org.acegisecurity.providers.ProviderManager.doAuthentication(ProviderManager.java:200)

at org.acegisecurity.AbstractAuthenticationManager.authenticate(AbstractAuthenticationManager.java:47)

at hudson.security.LDAPSecurityRealm$LDAPAuthenticationManager.authenticate(LDAPSecurityRealm.java:1019)

at jenkins.security.BasicHeaderRealPasswordAuthenticator.authenticate(BasicHeaderRealPasswordAuthenticator.java:56)

at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:79)

at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)

at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)

at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)

at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)

at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:90)

at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)

at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)

at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:51)

at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)

at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)

at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)

at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)

at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)

at jenkins.security.SuspiciousRequestFilter.doFilter(SuspiciousRequestFilter.java:36)

at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)

at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:549)

at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)

at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578)

at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)

at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)

at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)

at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)

at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1369)

at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)

at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:489)

at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)

at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)

at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1284)

at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)

at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)

at org.eclipse.jetty.server.Server.handle(Server.java:501)

at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)

at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)

at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)

at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:272)

at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)

at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)

at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)

at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)

at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)

at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)

at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)

at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)

at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)

at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)

at java.lang.Thread.run(Thread.java:748)

2021-04-25 02:20:46.328+0000 [id=17] SEVERE i.k.j.d.a.KubesphereApiTokenAuthenticator#authenticate: cannot get the review response or status is null by admin
```

**Comment by wansir** [comment]
Please paste the result of the following command `kubectl -n kubesphere-system exec -it openldap-0 slapcat`.

**Comment by wansir** [comment]
> Related Jenkins logs:
> 
> ```java
>  2021-04-25 02:19:55.909+0000 [id=26] INFO j.util.groovy.GroovyHookScript#execute: Executing /var/jenkins_home/init.groovy.d/initRBAC.groovy
> 
> Enabling role based authorisation strategy...
> 
> 2021-04-25 02:20:01.484+0000 [id=26] INFO o.c.j.p.k.KubernetesClientProvider$SaveableListenerImpl#onChange: Invalidating Kubernetes client: kubernetes null
> 
> Admin role created...OK
> 
> Read role created...OK
> 
> Saving changes.
> 
> 2021-04-25 02:20:01.705+0000 [id=26] INFO o.c.j.p.k.KubernetesClientProvider$SaveableListenerImpl#onChange: Invalidating Kubernetes client: kubernetes null
> 
> 2021-04-25 02:20:01.705+0000 [id=26] INFO j.util.groovy.GroovyHookScript#execute: Executing /var/jenkins_home/init.groovy.d/initSonarqube.groovy
> 
> --> SonarQube Disabled
> 
> 2021-04-25 02:20:04.091+0000 [id=26] INFO jenkins.InitReactorRunner$1#onAttained: Completed initialization
> 
> 2021-04-25 02:20:04.105+0000 [id=20] INFO o.c.j.p.k.KubernetesClientProvider$SaveableListenerImpl#onChange: Invalidating Kubernetes client: kubernetes null
> 
> 2021-04-25 02:20:04.216+0000 [id=20] INFO hudson.WebAppMain$3#run: Jenkins is fully up and running
> 
> 2021-04-25 02:20:31.931+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.tasks.Maven.MavenInstaller
> 
> 2021-04-25 02:20:32.405+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.plugins.sonar.MsBuildSonarQubeRunnerInstaller
> 
> 2021-04-25 02:20:32.680+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.plugins.sonar.SonarRunnerInstaller
> 
> 2021-04-25 02:20:32.919+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.tasks.Ant.AntInstaller
> 
> 2021-04-25 02:20:35.587+0000 [id=57] INFO h.m.DownloadService$Downloadable#load: Obtained the updated data file for hudson.tools.JDKInstaller
> 
> 2021-04-25 02:20:35.588+0000 [id=57] INFO hudson.util.Retrier#start: Performed the action check updates server successfully at the attempt #1
> 
> 2021-04-25 02:20:35.591+0000 [id=57] INFO hudson.model.AsyncPeriodicWork#lambda$doRun$0: Finished Download metadata. 45,499 ms
> 
> 2021-04-25 02:20:46.220+0000 [id=16] SEVERE i.k.j.d.a.KubesphereApiTokenAuthenticator#authenticate: cannot get the review response or status is null by admin
> 
> 2021-04-25 02:20:46.305+0000 [id=16] WARNING o.a.p.l.a.BindAuthenticator2#handleBindException: Failed to bind to LDAP: userDnuid=admin,ou=Users,dc=kubesphere,dc=io username=admin
> 
> javax.naming.AuthenticationException: [LDAP: error code 49 - Invalid Credentials]
> 
> at com.sun.jndi.ldap.LdapCtx.mapErrorCode(LdapCtx.java:3154)
> 
> at com.sun.jndi.ldap.LdapCtx.processReturnCode(LdapCtx.java:3100)
> 
> at com.sun.jndi.ldap.LdapCtx.processReturnCode(LdapCtx.java:2886)
> 
> at com.sun.jndi.ldap.LdapCtx.connect(LdapCtx.java:2800)
> 
> at com.sun.jndi.ldap.LdapCtx.<init>(LdapCtx.java:319)
> 
> at com.sun.jndi.ldap.LdapCtxFactory.getUsingURL(LdapCtxFactory.java:192)
> 
> at com.sun.jndi.ldap.LdapCtxFactory.getUsingURLs(LdapCtxFactory.java:210)
> 
> at com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxInstance(LdapCtxFactory.java:153)
> 
> at com.sun.jndi.ldap.LdapCtxFactory.getInitialContext(LdapCtxFactory.java:83)
> 
> at javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:684)
> 
> at javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:313)
> 
> at javax.naming.InitialContext.init(InitialContext.java:244)
> 
> at javax.naming.InitialContext.<init>(InitialContext.java:216)
> 
> at javax.naming.directory.InitialDirContext.<init>(InitialDirContext.java:101)
> 
> at org.acegisecurity.ldap.DefaultInitialDirContextFactory.connect(DefaultInitialDirContextFactory.java:180)
> 
> at org.acegisecurity.ldap.DefaultInitialDirContextFactory.newInitialDirContext(DefaultInitialDirContextFactory.java:261)
> 
> at org.acegisecurity.ldap.LdapTemplate.execute(LdapTemplate.java:123)
> 
> at org.acegisecurity.ldap.LdapTemplate.retrieveEntry(LdapTemplate.java:165)
> 
> at org.acegisecurity.providers.ldap.authenticator.BindAuthenticator.bindWithDn(BindAuthenticator.java:87)
> 
> at org.acegisecurity.providers.ldap.authenticator.BindAuthenticator.authenticate(BindAuthenticator.java:72)
> 
> at org.acegisecurity.providers.ldap.authenticator.BindAuthenticator2.authenticate(BindAuthenticator2.java:49)
> 
> at org.acegisecurity.providers.ldap.LdapAuthenticationProvider.retrieveUser(LdapAuthenticationProvider.java:233)
> 
> at org.acegisecurity.providers.dao.AbstractUserDetailsAuthenticationProvider.authenticate(AbstractUserDetailsAuthenticationProvider.java:122)
> 
> at org.acegisecurity.providers.ProviderManager.doAuthentication(ProviderManager.java:200)
> 
> at org.acegisecurity.AbstractAuthenticationManager.authenticate(AbstractAuthenticationManager.java:47)
> 
> at hudson.security.LDAPSecurityRealm$LDAPAuthenticationManager.authenticate(LDAPSecurityRealm.java:1019)
> 
> at jenkins.security.BasicHeaderRealPasswordAuthenticator.authenticate(BasicHeaderRealPasswordAuthenticator.java:56)
> 
> at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:79)
> 
> at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
> 
> at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
> 
> at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)
> 
> at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
> 
> at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:90)
> 
> at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
> 
> at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
> 
> at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:51)
> 
> at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
> 
> at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
> 
> at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
> 
> at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
> 
> at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
> 
> at jenkins.security.SuspiciousRequestFilter.doFilter(SuspiciousRequestFilter.java:36)
> 
> at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
> 
> at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:549)
> 
> at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
> 
> at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578)
> 
> at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
> 
> at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
> 
> at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)
> 
> at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
> 
> at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1369)
> 
> at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
> 
> at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:489)
> 
> at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)
> 
> at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
> 
> at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1284)
> 
> at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
> 
> at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
> 
> at org.eclipse.jetty.server.Server.handle(Server.java:501)
> 
> at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
> 
> at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)
> 
> at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
> 
> at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:272)
> 
> at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
> 
> at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
> 
> at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
> 
> at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
> 
> at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
> 
> at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
> 
> at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
> 
> at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
> 
> at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
> 
> at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
> 
> at java.lang.Thread.run(Thread.java:748)
> 
> 2021-04-25 02:20:46.328+0000 [id=17] SEVERE i.k.j.d.a.KubesphereApiTokenAuthenticator#authenticate: cannot get the review response or status is null by admin
> ```

I saw the error caused by ` Failed to bind to LDAP: userDnuid=admin,ou=Users,dc=kubesphere,dc=io username=admin`.

Data will be initialized after OpenLDAP started, make sure openldap is running, please wait a moment.

**Comment by JohnNiang** [comment]
> Please paste the result of the following command `kubectl -n kubesphere-system exec -it openldap-0 slapcat`.

Execution result:

```bash
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
dn: dc=kubesphere,dc=io
objectClass: top
objectClass: dcObject
objectClass: organization
o: kubesphere
dc: kubesphere
structuralObjectClass: organization
entryUUID: 0ffd1ccc-39b8-103b-949b-c59acb5a06d1
creatorsName: cn=admin,dc=kubesphere,dc=io
createTimestamp: 20210425021700Z
entryCSN: 20210425021700.594041Z#000000#000#000000
modifiersName: cn=admin,dc=kubesphere,dc=io
modifyTimestamp: 20210425021700Z

dn: cn=admin,dc=kubesphere,dc=io
objectClass: simpleSecurityObject
objectClass: organizationalRole
cn: admin
description: LDAP administrator
userPassword:: e1NTSEF9MHFiVHBZYTFrcXpXQXFjSTMyVmo0RWRpaC9nYytvNXc=
structuralObjectClass: organizationalRole
entryUUID: 0ffd4d32-39b8-103b-949c-c59acb5a06d1
creatorsName: cn=admin,dc=kubesphere,dc=io
createTimestamp: 20210425021700Z
entryCSN: 20210425021700.595297Z#000000#000#000000
modifiersName: cn=admin,dc=kubesphere,dc=io
modifyTimestamp: 20210425021700Z

dn: ou=Users,dc=kubesphere,dc=io
objectClass: organizationalUnit
objectClass: top
ou: Users
structuralObjectClass: organizationalUnit
entryUUID: 1097a256-39b8-103b-892e-d17b74c74dc2
creatorsName: cn=admin,dc=kubesphere,dc=io
createTimestamp: 20210425021701Z
entryCSN: 20210425021701.606730Z#000000#000#000000
modifiersName: cn=admin,dc=kubesphere,dc=io
modifyTimestamp: 20210425021701Z

dn: ou=Groups,dc=kubesphere,dc=io
objectClass: organizationalUnit
objectClass: top
ou: Users
ou: Groups
structuralObjectClass: organizationalUnit
entryUUID: 10980264-39b8-103b-892f-d17b74c74dc2
creatorsName: cn=admin,dc=kubesphere,dc=io
createTimestamp: 20210425021701Z
entryCSN: 20210425021701.609189Z#000000#000#000000
modifiersName: cn=admin,dc=kubesphere,dc=io
modifyTimestamp: 20210425021701Z

dn: uid=admin,ou=Users,dc=kubesphere,dc=io
objectClass: inetOrgPerson
objectClass: top
cn: admin
sn: admin
uid: admin
structuralObjectClass: inetOrgPerson
entryUUID: 109851c4-39b8-103b-8930-d17b74c74dc2
creatorsName: cn=admin,dc=kubesphere,dc=io
createTimestamp: 20210425021701Z
userPassword:: b3BlbmFkbWlu
entryCSN: 20210425033204.865117Z#000000#000#000000
modifiersName: cn=admin,dc=kubesphere,dc=io
modifyTimestamp: 20210425033204Z
```

BTW, the OLDAP-ha container log:

```bash
 Backing up /etc/ldap/slapd.d in /var/backups/slapd-2.4.50+dfsg-1~bpo10+1... done.

Creating initial configuration... done.

Creating LDAP directory... done.

invoke-rc.d: could not determine current runlevel

invoke-rc.d: policy-rc.d denied execution of restart.

config file testing succeeded

*** /container/run/startup/slapd failed with status 68
```

**Comment by JohnNiang** [comment]
Regresion fix by <https://github.com/kubesphere/ks-installer/pull/1443>.

**Comment by LinuxSuRen** [comment]
> Regresion fix by [kubesphere/ks-installer#1443](https://github.com/kubesphere/ks-installer/pull/1443).

I'm not sure about this. I believe this issue caused by the LDAP problem instead of the missing of openssl.
